quickie/src/main/kotlin/io/github/g00fy2/quickie/QROverlayView.kt

```diff
@@ -85,25 +85,36 @@ internal class QROverlayView @JvmOverloads constructor(
     super.onDraw(canvas)
   }
 
-  fun setCustomTextAndIcon(stringRes: Int, drawableRes: Int) {
+  fun setCustomText(stringRes: Int) {
     if (stringRes != 0) {
       try {
         binding.titleTextView.setText(stringRes)
       } catch (ignore: NotFoundException) {
         // string resource not found
       }
     }
+  }
+
+  fun setCustomIcon(drawableRes: Int?) {
     if (drawableRes != 0) {
       try {
-        ResourcesCompat.getDrawable(resources, drawableRes, null)?.limitDrawableSize()?.let {
-          binding.titleTextView.setCompoundDrawables(null, it, null, null)
+        if (drawableRes != null) {
+          ResourcesCompat.getDrawable(resources, drawableRes, null)?.limitDrawableSize()?.let {
+            setDrawableToTextView(it)
+          }
+        } else {
+          setDrawableToTextView(null)
         }
       } catch (ignore: NotFoundException) {
         // drawable resource not found
       }
     }
   }
 
+  private fun setDrawableToTextView(drawable: Drawable?) {
```

Comment:
Don't think this additional method is requiered. It gets only called in one place. :)